src/componentProvider.js

```diff
@@ -0,0 +1,70 @@
+import componentLoader from './componentLoader';
+
+/**
+ * Compile component elements and DOM attachments.
+ *
+ * @param {Object} config The component config.
+ * @returns {Function} A function to initialize the component.
+ */
+export default function componentProvider(config) {
+  const {
+    component: Component,
+    load, // default value omitted so `componentLoader` can use its default.
+    name,
+    querySelector = {},
+    querySelectorAll = {},
+  } = config;
+
+  if (typeof Component !== 'function') {
+    return undefined;
+  }
+
+  // The component selector.
+  const selector = `[data-component='${name}']`;
+
+  // Get component elements.
+  const componentEls = document.querySelectorAll(selector);
+
+  // Do nothing.
+  if (componentEls.length < 1) {
+    console.log(`No elements found for ${selector}`); // eslint-disable-line no-console
```

Comment:
I've always found this to be a nuisance in the old implementation. Particularly when you add the component in specific pages only, ex.: single.php, but bundle the component in a global file. It keeps "bothering" you everywhere the HTML element doesn't exist.

Reply:
I hear you. I see it as a hint that code that shouldn't be running on the page has been delivered and is executing, which is likely a performance issue. In this block-based world we live in, though, we may need a better/different bundling strategy. Something to think about!